### PR TITLE
Conditionally add HTTPS inbound allow firewall rule

### PR DESCRIPTION
**Pull request by strarsis**
This PR lets the firewall rule to allow inbound HTTPS to be added conditionally.
Only when at least one site uses HTTPS, the firewall rule for allowing inbound HTTPS is added.

### CODE REVIEW

File: roles/ferm/defaults/main.yml [hunk 1]
```diff
@@ -11,3 +11,5 @@ ferm_default_policy_forward: DROP
 ferm_input_list: []
 ferm_input_group_list: []
 ferm_input_host_list: []
+
+sites_using_ssl: "[{% for name, site in wordpress_sites.items() | list if site.ssl.enabled %}'{{ name }}',{% endfor %}]"
```
Comment (swalkinshaw) [inline roles/ferm/defaults/main.yml:15]: You should be able to re-use the existing `sites_use_ssl` here?

Comment (strarsis) [reply roles/ferm/defaults/main.yml]: Oh, I looked for such a variable, but had not found it. Re-using it is much better of course.
After some testing I noticed that just conditionally adding the allow rule would not make ansible clean it up if no site is using HTTPS (in subsequent runs).
Then I noticed the `delete` attribute of the individual ferm rules, which is ideal as it not only toggles the present of the HTTPS allow rule, but also cleans it up if it should not apply anymore (as no sites are using SSL).

**Comment by strarsis** [comment]
Edit: Using the `delete` attribute as this allows toggling _and_ proper clean up of file when HTTPS firewall allow rule is not needed (in subsequent provisions), it also allows for much simpler configuration code.

**Comment by swalkinshaw** [comment]
Thanks @strarsis

**Pull request merged**